webapp/components/notifications/NotificationMenu/NotificationMenu.vue

```diff
@@ -46,24 +48,53 @@ export default {
           variables,
         })
         if (!(markAsRead && markAsRead.read === true)) return
-        this.notifications = this.notifications.map(n => {
-          return n.from.id === markAsRead.from.id ? markAsRead : n
+        this.displayedNotifications = this.displayedNotifications.map(n => {
+          return this.equalNotification(n, markAsRead) ? markAsRead : n
         })
       } catch (err) {
-        throw new Error(err)
+        this.$toast.error(err.message)
       }
     },
+    equalNotification(a, b) {
+      return a.from.id === b.from.id && a.createdAt === b.createdAt && a.reason === b.reason
+    },
   },
   computed: {
-    totalNotifications() {
-      return (this.notifications || []).length
+    notificationsCount() {
+      return (this.displayedNotifications || []).length
+    },
+    unreadNotificationsCount() {
+      let countUnread = 0
```

Comment:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-09-23T10:59:50Z" title="Monday, September 23rd 2019, 12:59:50 pm +02:00">Sep 23, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -58,6 +104,15 @@ export default {
     totalNotifications() {
       return (this.notifications || []).length
     },
+    unredNotifications() {
+      let countUnread = 0
```


```suggestion
      let countUnread = this.notifications.reduce((notification, counter) => {
        return (notification.read) ? counter : counter +1
      })
```

See:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/Reduce

Reply:
> [<img alt="Tirokk" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/Tirokk) **Authored by [Tirokk](https://github.com/Tirokk)**
_<time datetime="2019-09-23T11:46:03Z" title="Monday, September 23rd 2019, 1:46:03 pm +02:00">Sep 23, 2019</time>_

---


I like this ! 😍

Reply:
> [<img alt="Tirokk" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/Tirokk) **Authored by [Tirokk](https://github.com/Tirokk)**
_<time datetime="2019-09-23T14:02:03Z" title="Monday, September 23rd 2019, 4:02:03 pm +02:00">Sep 23, 2019</time>_

---


But this does not function at all.
Don’t know why and will not spend hours of investigating for saving not on single line and add a bit of complexity …

Reply:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-09-26T06:54:47Z" title="Thursday, September 26th 2019, 8:54:47 am +02:00">Sep 26, 2019</time>_

---


Sorry, I reversed the `accumulator` and `currentValue` from the example and I forgot to add the 0 as a start value.

This should work:
```js
[{read: true}, {read: false}].reduce((counter, notification) => {
  return (notification.read) ? counter : counter +1
}, 0)
```

Reply:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-09-26T07:45:53Z" title="Thursday, September 26th 2019, 9:45:53 am +02:00">Sep 26, 2019</time>_

---


this might very well work @roschaefer, but does it really reduce complexity? 
`KISS`?

Reply:
> [<img alt="Tirokk" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/Tirokk) **Authored by [Tirokk](https://github.com/Tirokk)**
_<time datetime="2019-09-26T09:07:53Z" title="Thursday, September 26th 2019, 11:07:53 am +02:00">Sep 26, 2019</time>_

---


This adds kind of inner complexity in my eyes and is bad understandable.
Like @mattwr18 I say `KISS`!

Reply:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-09-26T12:57:39Z" title="Thursday, September 26th 2019, 2:57:39 pm +02:00">Sep 26, 2019</time>_

---


I thought a `fold` is not so complex but I might be wrong :neutral_face: 

Here's the concept on Wikipedia: https://en.wikipedia.org/wiki/Fold_(higher-order_function)